### PR TITLE
add optional ingress for vmagent/vmalert/vmsingle

### PR DESCRIPTION
**Pull request by kevinvirs**
update dependencies version to latest
add optional ingress for vmagent/vmalert/vmsingle
change alertmanager ingress temlate

### CODE REVIEW

**Review by k1rk — CHANGES_REQUESTED** [review]
thanks for contribution

left some comments regarding some empty lines and trim

File: charts/victoria-metrics-k8s-stack/templates/victoria-metrics-operator/alertmanager/ingress.yaml [hunk 1]
```diff
@@ -1,64 +1,81 @@
...
-  name: {{ printf "%s-%s" (include "victoria-metrics-k8s-stack.fullname" $) "alertmanager" | trunc 63 | trimSuffix "-" }}
+  name: {{ .Values.alertmanager.name | default $serviceName }}
   namespace: {{ .Release.Namespace }}
-{{- if .Values.alertmanager.ingress.annotations }}
-  annotations:
-{{ toYaml .Values.alertmanager.ingress.annotations | indent 4 }}
-{{- end }}
   labels:
     app.kubernetes.io/component: {{ include "victoria-metrics-k8s-stack.name" $ }}-alertmanager
+{{ include "victoria-metrics-k8s-stack.labels" . | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (kevinvirs) [reply charts/victoria-metrics-k8s-stack/templates/victoria-metrics-operator/alertmanager/ingress.yaml]: fixed

File: charts/victoria-metrics-k8s-stack/templates/victoria-metrics-operator/alertmanager/ingress.yaml [hunk 2]
```diff
@@ -1,64 +1,81 @@
 {{- if and .Values.alertmanager.enabled .Values.alertmanager.ingress.enabled }}
-{{- $pathType := .Values.alertmanager.ingress.pathType | default "" }}
-{{- $serviceName := printf "%s-%s" "vmalertmanager" (include "victoria-metrics-k8s-stack.fullname" .) }}
 {{- $servicePort := 9093 -}}
-{{- $paths := compact ( append ( .Values.alertmanager.ingress.paths | default ( list "" )) .Values.alertmanager.spec.routePrefix) -}}
-{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1" }}
+{{- $ingressPath := .Values.alertmanager.ingress.path -}}
+{{- $serviceName := printf "%s-%s" "alertmanager" (include "victoria-metrics-k8s-stack.fullname" .) }}
```
[the remark this answers is not in the document]
Comment (kevinvirs) [reply charts/victoria-metrics-k8s-stack/templates/victoria-metrics-operator/alertmanager/ingress.yaml]: fixed

File: charts/victoria-metrics-k8s-stack/templates/victoria-metrics-operator/alertmanager/ingress.yaml [hunk 3]
```diff
@@ -1,64 +1,81 @@
...
+  {{- if .Values.alertmanager.ingress.hosts  }}
+  {{- range .Values.alertmanager.ingress.hosts }}
+    - host: {{ tpl . $}}
       http:
         paths:
-  {{- range $p := $paths }}
-          - path: {{ tpl $p $ }}
-            {{- if $pathType }}
-            pathType: {{ $pathType }}
+{{ if $extraPaths }}
```
[the remark this answers is not in the document]
Comment (kevinvirs) [reply charts/victoria-metrics-k8s-stack/templates/victoria-metrics-operator/alertmanager/ingress.yaml]: fixed

File: charts/victoria-metrics-k8s-stack/templates/victoria-metrics-operator/alertmanager/ingress.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevinvirs) [reply charts/victoria-metrics-k8s-stack/templates/victoria-metrics-operator/alertmanager/ingress.yaml]: fixed

**Review by kevinvirs — COMMENTED** [review]
fix requested
tested on kube 1.18

**Review by k1rk — APPROVED** [review]
LGTM! 

@kevinvirs thanks

**Pull request merged**